src/Symfony/Component/Form/Extension/Core/Type/DateTimeType.php

```diff
@@ -107,7 +107,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'invalid_message_parameters',
             ]));
 
-            if (isset($emptyData['date'])) {
+            if ($emptyData instanceof \Closure) {
+                $lazyEmptyData = static function ($option) use ($emptyData) {
```

Comment:
I was a bit confused why we would define a closure here if we just call it one the next line until I realised that it is reused later on. I think we should move the closure declaration outside the `if` block to avoid this confusion. Having it being used outside the block it is defined in should be avoided.